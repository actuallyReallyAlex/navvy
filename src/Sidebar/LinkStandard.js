import React from 'react'
import PropTypes from 'prop-types'
import { Box, Button, Text } from 'grommet'
import LinkIndicator from './LinkIndicator'
import LinkIcon from './LinkIcon'

const LinkStandard = ({
  handler,
  icon,
  selected,
  setSelected,
  size,
  title
}) => {
  const handleClick = () => {
    setSelected(title)
    handler && handler()
  }

  return (
    <Button hoverIndicator onClick={handleClick} plain>
      <Box
        direction="row"
        height="45px"
        margin={{ vertical: 'xsmall', left: 'xsmall' }}
        pad={{ vertical: 'small', left: 'small' }}
        style={{ position: 'relative' }}
        width="95%"
      >
        {selected === title && <LinkIndicator />}
        <Box
          align="center"
          direction="row"
          gap="small"
          justify={size === 'expanded' ? 'start' : 'center'}
          width={size === 'expanded' ? '100%' : '75%'}
        >
          <LinkIcon icon={icon} />
          {size === 'expanded' && <Text>{title}</Text>}
        </Box>
      </Box>
    </Button>
  )
}

LinkStandard.propTypes = {
  handler: PropTypes.func,
  icon: PropTypes.string,
  selected: PropTypes.string.isRequired,
  setSelected: PropTypes.func.isRequired,
  size: PropTypes.oneOf(['expanded', 'collapsed']).isRequired,
  title: PropTypes.string.isRequired
}

export default LinkStandard
