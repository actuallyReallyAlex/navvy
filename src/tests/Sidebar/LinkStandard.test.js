import React from 'react'
import LinkStandard from '../../Sidebar/LinkStandard'
import { shallow } from 'enzyme'

const defaultProps = {
  handler: jest.fn(),
  icon: 'Cube',
  selected: 'Selected',
  setSelected: jest.fn(),
  size: 'expanded',
  title: 'Title'
}

describe('<LinkStandard />', () => {
  afterEach(() => {
    defaultProps.handler.mockClear()
    defaultProps.setSelected.mockClear()
  })

  test('Should render the <LinkStandard /> component.', () => {
    const component = shallow(<LinkStandard {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
