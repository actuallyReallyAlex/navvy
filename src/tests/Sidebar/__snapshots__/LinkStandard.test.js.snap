// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkStandard /> Should render the <LinkStandard /> component. 1`] = `ShallowWrapper {}`;
